packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -272,35 +272,104 @@ class OverlayState extends State<Overlay> with TickerProviderStateMixin {
     insertAll(widget.initialEntries);
   }
 
+  int _insertionIndex(OverlayEntry below, OverlayEntry above) {
+    assert(above == null || below == null);
+    if (below != null)
+      return _entries.indexOf(below);
+    if (above != null)
+      return _entries.indexOf(above) + 1;
+    return _entries.length;
+  }
+
   /// Insert the given entry into the overlay.
   ///
+  /// If `below` is non-null, the entry is inserted just below `below`.
   /// If `above` is non-null, the entry is inserted just above `above`.
   /// Otherwise, the entry is inserted on top.
-  void insert(OverlayEntry entry, { OverlayEntry above }) {
-    assert(entry._overlay == null);
+  ///
+  /// It is an error to specify both `above` and `below`.
```

Comment:
What would you think about instead creating a new method called `insertBelow`?  I'm wondering if it will be confusing to developers using this in an IDE who may think they have to specify both optional parameters, or who will try to specify both and not realize their error until runtime.

Reply:
We use that pattern in other places of the framework as well, e.g. https://github.com/flutter/flutter/blob/387f885481cd8441237680de0cfcd0802c0af48c/packages/flutter/lib/src/painting/text_style.dart#L328-L329.

And I think it's actually a nice and concise way to specify an API without convoluting the name space.

Reply:
In that case, can we add messages to the new asserts?

Reply:
Done. PTAL